Fix React Native createObjectURL polyfill incompatibility

nitro404 opened the pull request:
React Native introduces its own polyfill for window.URL.createObjectURL which is not compatible with Shaka. Encapsulating a reference to the original function inside of the MediaSourceEngine circumvents this issue.

Issue #1842

nitro404 commented:
I signed the CLA.

joeyparrish commented:
@nitro404, can you please fix the linter errors?  You can check this for yourself by running `python build/all.py` before pushing your commits.

nitro404 commented:
Should be fixed! I'm unfortunately not able to run "test.py" on my current environment, the script fails with an error, likely because I have Python 3 installed.

joeyparrish commented:
Ah, yeah, those scripts currently require python2.  If you have that, you can run `python2 build/test.py`.

But I'll run it through the build bot again.

joeyparrish commented:
For variables you reassign, use `let` instead of `const`.  For the private access issue, you can make the member public (remove the `@private` annotation and the trailing underscore) and make it clear in the comments that it is for use by tests only.

nitro404 commented:
Ahh, sorry! I've adjusted my environment and fixed all outstanding issues. Updated the function description and re-based onto master as well. Let me know if any other changes are needed, thank you!

joeyparrish commented:
Thanks!  I've requested another test run through our build bot.

joeyparrish commented:
Thank you for your contribution!
